ansys/mapdl/core/licensing.py

```diff
@@ -0,0 +1,456 @@
+
+    Raises
+    ------
+    LicenseServerConnectionError
+        If there is an error message in the license log file.
+
+    """
+    licdebug_file = os.path.join(get_licdebug_path(), get_licdebug_name())
+    file_iterator = get_licdebug_msg(licdebug_file)
+
+    max_time = time.time() + timeout
+    while max_time < time.time():
+        msg = next(file_iterator)
+
+        if "DENIED" in msg:
+            license_path = re.findall("(?<=License path:)(.*)(?=;\n)", msg)[0]
+            license_port = license_path.split("@")[0]
+            license_hostname = license_path.split("@")[1]
+            raise LicenseServerConnectionError(
+                head_message=f"Error connecting to {license_port}:{license_hostname}",
+                error_message=msg,
+                tail_message=f"Error found in file {get_licdebug_name()}",
+            )
+        if 'CHECKOUT            ' + APP_NAME in msg:
+            # successfully license checkout
+            break
+
+
+def get_licdebug_path():
+    """Get license client log (``licdebug``) path.
+
+    This path is obtained from the correspondent env variable (OS dependent) and appending ``.ansys``.
+
+    Returns
+    -------
+    str
+        path of the license client log file.
+    """
+    if os.name == "nt":
+        folder = os.getenv("TEMP")
+    elif os.name == "posix":
+        folder = os.getenv("HOME")
+    else:
+        raise OSError(f"Unsupported OS {os.name}")
+
+    return os.path.join(folder, ".ansys")
+
+
+def get_licdebug_name():
+    """Get license client log file name.
+
+    This file change the name according to the ANSYS version and the type of license requested (``$appname``).
+    * For ANSYS version 22.1 and above: `licdebug.$hostname.$appname.$version.out`
+    * For ANSYS version 21.2 and below: `licdebug.$appname.$version.out`
+
+    where:
+    * ``$hostname`` is the name of the machine.
+    * ``$appname`` is the name of the feature used by the license client. Eg. 'FEAT_ANSYS'
+    * ``$version`` is the version of ANSYS. Eg 211 for version 21.1.
+
+    Returns
+    -------
+    str
+        licdebug log file complete name.
+
+    """
+    # Licdebug name convention:
+    # - For version 22.1 and above: `licdebug.$hostname.$appname.$version.out`
+    # - For version 21.2 and below: `licdebug.$appname.$version.out`
+
+    from ansys.mapdl.core.launcher import _version_from_path, get_ansys_path
+
+    name = "licdebug"
+    hostname = socket.gethostname()
+    appname = APP_NAME
+    # This is the type of license my client requests (Windows 10, 2021R2)
+    version = _version_from_path(get_ansys_path(allow_input=False))
+    ending = "out"
+
+    if version < 221:
+        parts = (name, appname, version, ending)
+    else:
+        parts = (name, hostname, appname, version, ending)
+
+    return ".".join([str(each_part) for each_part in parts])
+
+
+def get_licdebug_msg(licdebug_file):
```

Comment:
Nitpick: This is more of a generalized method that can be applied to any file, and it resembles the `tail` method for Linux (I'm sure there's a windows counterpart as well.  I'd just rename it as `tail` and most developers will know what you mean

Reply:
I know what you mean. We have that in Pandas (``df.tail()``) to show the last bits of a dataframe. 

Changing to `get_licdebug_tail`.

Reply:
Perfect.